pkg/typesystem/typesystem.go

```diff
@@ -820,7 +825,7 @@ func hasEntrypoints(
 		return true, false, nil
 	}
 
-	return false, false, nil
```

Comment:
this line is suspected to be unreachable, hence why i am writing a panic, because i can't figure out a test for it.

Reply:
I think this was originally here because of the distaste for panics in the code, but this is a perfect usage of them IMO. Thanks for the update.